lib/driver.md

```diff
@@ -0,0 +1,374 @@
+    rule unparsePattern(\ceil { S1 , S2 } (P1))       => "\\ceil{" +String unparseSort(S1) +String "," +String unparseSort(S2)  +String "} (" +String unparsePattern(P1) +String ")"
+    rule unparsePattern(\not { S1 } (P1))             => "\\not{" +String unparseSort(S1) +String "} (" +String unparsePattern(P1) +String ")"
+    rule unparsePattern(S(Args:Patterns))             => unparseSymbol(S) +String "(" +String unparsePatterns(Args) +String ")"
+    rule unparsePattern(\and { S1 } (P1, P2))
+      => "\\and{" +String unparseSort(S1) +String "} (" +String unparsePatterns(P1) +String "," +String unparsePatterns(P2) +String  ")"
+    rule unparsePattern(\or { S1 } (P1, P2))
+      => "\\or{" +String unparseSort(S1) +String "} (" +String unparsePatterns(P1) +String "," +String unparsePatterns(P2) +String  ")"
+    rule unparsePattern(\forall  { S1 } (P1, P2)) => "\\forall {" +String unparseSort(S1) +String "} (" +String unparsePattern(P1) +String " , " +String unparsePattern(P2) +String ")"
+    rule unparsePattern(\exists  { S1 } (P1, P2)) => "\\exists {" +String unparseSort(S1) +String "} (" +String unparsePattern(P1) +String " , " +String unparsePattern(P2) +String ")"
+
+    syntax String ::= KoreNameToString(KoreName) [function, functional, hook(STRING.token2string)]
+
+    syntax String ::= unparseSort(Sort) [function, functional]
+    rule unparseSort(KoreName {}) => KoreNameToString(KoreName) +String "{}"
+
+    syntax String ::= unparseSymbol(Symbol) [function, functional]
+    rule unparseSymbol(KoreName {Sorts}) => KoreNameToString(KoreName) +String "{" +String unparseSorts(Sorts) +String "}"
+
+    syntax String ::= unparsePatterns(Patterns) [function, functional]
+    rule unparsePatterns(P, Ps) => unparsePattern(P) +String "," +String unparsePatterns(Ps) requires notBool Ps ==K .Patterns
+    rule unparsePatterns(P, .Patterns) => unparsePattern(P)
+    rule unparsePatterns(.Patterns) => ""
+
+    syntax String ::= unparseSorts(Sorts) [function, functional]
+    rule unparseSorts(S, Ss) => unparseSort(S) +String "," +String unparseSorts(Ss) requires notBool Ss ==K .Sorts
+    rule unparseSorts(S, .Sorts) => unparseSort(S)
+    rule unparseSorts(.Sorts) => ""
+endmodule
+```
+
+`KORE-PARSE`
+============
+
+```k
+module KORE-PARSE
+    imports IO-NONFUNCTIONAL
+    imports KORE
+    imports K-REFLECTION
+
+    syntax PrePattern ::= Pattern
+
+    syntax PrePattern ::= parse(input: PreString, parser: String)
+                        | parseFile(filename: PreString, parser: String) [seqstrict(1), result(String)]
+    rule parse(Program, Parser) => parseFile(writeTempFile(Program), Parser)
+    rule parseFile(File, Parser) => parseKore(system(Parser +String " " +String File))
+
+    syntax PrePattern ::= parseKore(PreString) [seqstrict(1), result(String)]
+    rule parseKore(String) => #parseKORE(String):Pattern
```

Comment:
For my curiosity, what happens when this fails to be in the `Pattern` sort? We should find out.

Hypothesis: it will create an incorrect injection and get stuck.